send-sms-install-link/send-sms-install-link.js

```diff
@@ -4,29 +4,48 @@ module.exports = {
   name: 'Should the user see the "Send install link via SMS" screen',
   hypothesis: 'Allowing users to send a Firefox Mobile link via SMS will result in a better conversion rate',
   startDate: '2017-01-01',
-  subjectAttributes: ['account'],
+  subjectAttributes: ['account', 'forceExperiment', 'forceExperimentGroup', 'uniqueUserId'],
   independentVariables: ['sendSms'],
   eligibilityFunction: function (subject) {
-    if (! subject || ! subject.account) {
+    if (! subject || ! subject.account || ! subject.uniqueUserId) {
       return false;
     }
 
-    var email = subject.account.get('email') || '';
-    if (/@softvision\.com$/.test(email)) {
-      return true;
-    } else if (/@mozilla\.(com|org)$/.test(email)) {
+    if (isInExperiment(subject.forceExperiment, subject.account.get('email'))) {
       return true;
     }
 
-    return false;
+    // a random sampling of 5% of all sessions.
+    // 2.5% will be in the control group,
+    // 2.5% in the treatment group.
+    return this.bernoulliTrial(0.05, subject.uniqueUserId);
+
+    function isInExperiment(forceExperiment, email) {
+      return forceExperiment === 'sendSms' ||
+             // The regexps are duplicated in the groupingFunction because
+             // there is no way to share code between the two.
```

Comment:
😯 ...? Aren't they just closures like anything else? Are they parsed independently or something? I'd quite like to undestand as this repo seems completely opaque and magical to me at the moment...